core/domain/feedback_domain.py

```diff
@@ -37,16 +37,32 @@ class FeedbackThread(python_utils.OBJECT):
         summary: str. A summary of the feedback thread.
         has_suggestion: bool. Whether the feedback thread includes a
             suggestion.
+        message_count: int. The number of messages posted onto the thread.
         created_on: datetime.datetime. The time when the feedback thread was
             created.
         last_updated: datetime.datetime. The time when the feedback thread
             was last updated.
+        updated_status: str. The change in status since the last message, if
```

Comment:
This seems odd as a field in the thread. Perhaps you mean "current_status" or just "status"? But in that case I think it duplicates the existing "status" field.

Reply:
So it means the change made by the last message, if any, to the status of the thread (which requires details about both the last and second-to-last messages).

I think it's fine, although I'm fine with renaming it if you have any better choices.

Reply:
But why do we need it in the thread model? Wouldn't the existing `status` field suffice?

Reply:
Not entirely, the messages we show on the UI have an indicator for when a message changes, for example:

> *Status changed to: closed*

Since this indicator should only appear when the status actually changes, I've added this field to make that process simpler.

Reply:
Ah, I see.

I suggest a different way to handle this, though. How about keeping the "status" field, and changing the other field to be more like "last nonempty message"? That way you can show the latest message on the thread and its current status, which I think would make more sense to me as the thread's "current state".

Reply:
Well I'd still like to show the status change in the event a message is non-empty, I think that's still be helpful?

> **Last Message:**
> OK I need to do more research, thanks for input!
> *Status changed to Open*

Reply:
Hm, I disagree -- I would want to know what the "current state" of the thread is, and would prioritize showing that quite a bit more over the last "state-change action" that someone happens to have taken. We could show both perhaps, but I think that's not really necessary TBH, since it isn't really much of a sneak preview of useful info in the thread.

Reply:
Hmm, well given it's a property of the message I think maybe it'd be fine if we rename it to `last_message_updated_status` instead, to point that out? I still like the idea of showing both because we still show the same status-change information on the thread itself, so it'd be inconsistent to have it missing from the task "preview"

Reply:
I think the preview is of the "thread" as a whole, not the latest message, so the requirements for what to show are actually a bit different. Each message is basically a "status update/change" to the thread, but I'm trying to find a way to summarize "here's where the thread is now" for the thread summary display. That's why I think we want to show the current thread status and the last nonempty message, since those seem (from a product perspective) like a good way to convey that information. The last-message-updated-status -- eh, not so much.

(Does this make sense? I was hoping to discuss this in the meeting today actually...)

Reply:
Hmm well right now all we have is the bubble indicator in the top-right corner, so I'll concede and leave that as the primary notification and render a status-change signal whenever the `last_message_text` is empty.

Thanks!